presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/InMemoryHashAggregationBuilder.java

```diff
@@ -348,14 +393,33 @@ public Type getType()
 
         public void processPage(GroupByIdBlock groupIds, Page page)
         {
-            if (step.isInputRaw()) {
+            if (!this.orderByChannels.isEmpty()) {
+                checkArgument(step == Step.SINGLE && pagesIndex.isPresent(), "ORDER BY in aggregations is supported only for single step aggregations");
+                pagesIndex.ifPresent(index -> index.addPage(page));
```

Comment:
No, but I'd get a warning in IntelliJ if I do pagesIndex.get(). I think checkArgument messed up with the analysis.